Kconfig: Add missing symbols NRFX_UARTE[2,3] and SPI_3_NRF_SPIS

anangl opened the pull request:
**drivers: spi: nrfx: Add support for SPIS3**

This patch adds missing SPI_3_NRF_SPIS Kconfig option (referenced by
`ext/hal/nordic/nrfx_config_nrf9160.h`, although not defined anywhere)
and adds support for SPIS3 instance in the nrfx_spis driver shim.

Fixes #13750.

**ext: nordic: Add NRF_SPIS[0-3] definitions to nrfx_config_nrf9160.h**

These entries should have been added in the initial version of
this file (in commit 99bab3f)
but their absence was overlooked.

**ext: nordic: Add missing NRFX_UARTE[2,3] Kconfig options**

Definitions of these two options were missing, altought the symbols
that would be created for them were already referenced from
`ext/hal/nordic/nrfx_config_nrf9160.h`.

Fixes #13747.

anangl committed:
drivers: spi: nrfx: Add support for SPIS3

This patch adds missing SPI_3_NRF_SPIS Kconfig option (referenced by
`ext/hal/nordic/nrfx_config_nrf9160.h`, although not defined anywhere)
and adds support for SPIS3 instance in the nrfx_spis driver shim.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
ext: nordic: Add missing NRFX_UARTE[2,3] Kconfig options

Definitions of these two options were missing, altought the symbols
that would be created for them were already referenced from
`ext/hal/nordic/nrfx_config_nrf9160.h`.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
ext: nordic: Add NRF_SPIS[0-3] definitions to nrfx_config_nrf9160.h

These entries should have been added in the initial version of
this file (in commit 99bab3f478a0b578549fcb4f13f88ea00baadd73)
but their absence was overlooked.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

ioannisg commented on drivers/spi/Kconfig.nrfx:
```diff
@@ -162,13 +162,29 @@ endif # SPI_2
 
 if SPI_3
 
+choice
```
Question: why this is not symmetric with the rest of SPI ports?
E.g. there is no config SPI_3_NRF_SPI

anangl replied:
Because there is no such peripheral in any of our SoCs. There are SPIs 0-2 in nRF51/52, and in nRF91 there are only SPIMs. I don't think it is worth adding such entry just for symmetry.

ioannisg commented on drivers/spi/Kconfig.nrfx:
```diff
@@ -162,13 +162,29 @@ endif # SPI_2
 
 if SPI_3
```
I am confused with the symmetry, here:
Why we have if `SPI_2 `and if `SPI_3 `but `if SPI_1 && !I2C_1`

anangl replied:
Yes, such checks should be added. They were added to `i2c/Kconfig.nrfx` but not here. I'll fix this.

ioannisg replied:
Thanks.

ioannisg reviewed (commented):
Some questions for clarification, only, otherwise looks OK.

anangl committed:
drivers: nrfx: Add Kconfig checks for mutually exclusive peripherals

In Nordic SoCs, SPI and TWI peripherals with the same instance number
share certain resources and therefore cannot be used at the same time
(in nRF91 Series this limitation concerns UART peripherals as well).
This patch adds Kconfig checks ensuring that only one of such mutually
exclusive peripherals can be enabled.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
drivers: spi: nrfx: Improve a few Kconfig option prompts

Rephrase prompts for a few options in drivers/spi/Kconfig.nrfx
so that their purpose is presented a bit better.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented:
@ioannisg I addressed your comments.

ioannisg reviewed (commented):
Thanks for addressing the issues; I have a last, related comment:
Since you went through and added the mutual exclusive conditionals for SPIs and TWIs, I wonder if we should do the same for UARTEs; E.g. in nRF9160, UARTEx, SPIx and I2Cx are all on the same line...

anangl commented:
@ioannisg
> Since you went through and added the mutual exclusive conditionals for SPIs and TWIs, I wonder if we should do the same for UARTEs; E.g. in nRF9160, UARTEx, SPIx and I2Cx are all on the same line...

I included UARTE peripherals in these checks for nRF9160.

ioannisg commented on drivers/spi/Kconfig.nrfx:
```diff
@@ -111,9 +115,12 @@ config SPI_1_NRF_ORC
-if SPI_2
+# In Nordic SoCs, SPI and TWI peripherals with the same instance number
+# share certain resources and therefore cannot be used at the same time
+# (in nRF91 Series this limitation concerns UART peripherals as well).
```
nit: UARTE, not UART

anangl replied:
Actually, it was intentional. I wanted to refer to UART as a general type of peripheral, like I did above with SPI and TWI (without going into details with SPIM, SPIS and so on). I was a bit afraid that this may appear ambiguous but I couldn't find a more accurate phrasing - in case I used UARTE here, the SPI and TWI above might be understood as referring to only the legacy kind of Nordic peripherals. And on the other hand, UARTE is still a UART (Universal asynchronous receiver/transmitter) with EasyDMA. Even the Product Specification sometimes uses the UART abbreviation for it, e.g. `TASKS_STARTRX` is described as "Start UART receiver".

ioannisg commented on drivers/spi/Kconfig.nrfx:
```diff
@@ -158,43 +165,63 @@ config SPI_2_NRF_ORC
-if SPI_3
+# In Nordic SoCs, SPI and TWI peripherals with the same instance number
+# share certain resources and therefore cannot be used at the same time
+# (in nRF91 Series this limitation concerns UART peripherals as well).
```
...and here
